watson/cli.py

```diff
@@ -1053,6 +1071,25 @@ def add(watson, args, from_, to, confirm_new_project, confirm_new_tag):
             confirm_new_tag):
         confirm_tags(tags, watson.tags)
 
+    # check if <from> and <to> are time without a date, if yes
+    # convert to todays datetime
+    if isDateTime(from_) and isDateTime(to):
+        from_ = arrow.get(from_).replace(tzinfo='local')
+        to = arrow.get(to).replace(tzinfo='local')
+    elif isDateTime(from_) and isTime(to):
+        from_ = arrow.get(from_).replace(tzinfo='local')
+        to = getMergedDateTime(from_.floor("day"), to)        
+    elif isTime(from_) and isTime(to):
+        from_ = getDateTimeToday(from_)
+        to = getDateTimeToday(to)
```

Comment:
I wonder if Arrow doesn't provide something to help with that. It feels a bit tedious to introduce our own regexp to parse the date & times.

Reply:
I have checked the api and didn't find s.th. appropriate. But I am open for suggestions.

Reply:
I've been looking into it and there's something I don't understand: why are there two different click types `Date` and `Time` in `cli.py`?

It turns out that both accept the full date, but using different formats!

* `TimeParamType` (aka, `Time`) is only used in `watson stop`:
```
$ watson stop --at 09:00
$ watson stop --at 2019-05-23T12:00
```
* `DateParamType` (aka, `Date`) is used un every other place where a date is required:
```
$ watson report -f 2019-05-23
$ watson report -f "2019-05-23 12:00"
```

In my opinion, the following changes might solve issue #278 in a simpler way:
1. Rename `Date` to `DateTime` which is more appropriate to what it does. Of course, rename also the corresponding `ParamType` classes.
2. Use only `DateTime` to unify datetime input management for the user. This means `watson stop --at` date format changes, but I think it's OK.
3. Allow `DateTime` to receive time only dates, which is precisely what the `Time` type does. You could reuse the code.
4. Remove `Time` and `TimeParamType`.

Please, remember that I'm not a reviewer of this project, so I might be missing important stuff!! :-) what do you think @k4nar?

Reply:
I like the idea. I didn't dive deep so I was not aware of TimeParamType which is basically doing what I implemented to handle time only.
I also couldn't agree more that the api is not consistent for the different cmds and it felt like adding another balcony.
The only drawback I see so far is that it will be hard to handle the case where we want to reuse the date specified for --from for --to as the parameter processing would be done without the knowledge about the other parameter.